gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/LaunchDagTask.java

```diff
@@ -36,4 +43,25 @@ public LaunchDagTask(DagActionStore.DagAction dagAction, LeaseAttemptStatus.Leas
   public <T> T host(DagTaskVisitor<T> visitor) {
     return visitor.meet(this);
   }
+
+  @Override
+  public final boolean conclude() {
+    try {
+      // Delete adhoc flowSpecs from catalog if the dag was concluded properly
+      if (super.conclude()) {
+        DagManager.DagId dagId = DagManagerUtils.generateDagId(this.dagAction.getFlowGroup(),
+            this.dagAction.getFlowName(), this.dagAction.getFlowExecutionId());
+        FlowSpec flowSpec =
+            this.dagManagementStateStore.getFlowSpec(FlowSpec.Utils.createFlowSpecUri(dagId.getFlowId()));
+        flowSpec.addProperty(ConfigurationKeys.FLOW_EXECUTION_ID_KEY, dagId.getFlowExecutionId());
+        if (!flowSpec.isScheduled()) {
+          dagManagementStateStore.removeFlowSpec(flowSpec);
```

Comment:
we can use `dagManagementStateStore.removeFlowSpec(flowSpec.getUri(), new Properties(), false);` and then there is no need for that API, right?
I prefer the first one, because that matches FLowCatalog API signature. But if you decide to use this one then we can delete `void removeFlowSpec(URI uri, Properties headers, boolean triggerListener)` from DMSS

Reply:
My take is that I prefer avoiding overloading functions unless absolutely needed

Reply:
I prefer removing unused APIs/code, if there is no definitive plan to use it in near future.

Reply:
Okay removing the overloaded function and we can use provide the default values